src/lib/operators/table_scan/column_is_null_table_scan_impl.cpp

```diff
@@ -26,36 +28,43 @@ std::string ColumnIsNullTableScanImpl::description() const {
   return "IsNullScan";
 }
 
-std::shared_ptr<RowIDPosList> ColumnIsNullTableScanImpl::scan_chunk(const ChunkID chunk_id) {
-  const auto& chunk = _in_table->get_chunk(chunk_id);
-  const auto& segment = chunk->get_segment(_column_id);
-
-  auto matches = std::make_shared<RowIDPosList>();
-
-  if (const auto value_segment = std::dynamic_pointer_cast<BaseValueSegment>(segment)) {
-    _scan_value_segment(*value_segment, chunk_id, *matches);
-  } else {
-    const auto& chunk_sorted_by = chunk->individually_sorted_by();
-    if (!chunk_sorted_by.empty()) {
-      for (const auto& sorted_by : chunk_sorted_by) {
-        if (sorted_by.column == _column_id) {
-          _scan_generic_sorted_segment(*segment, chunk_id, *matches, sorted_by.sort_mode);
-          ++num_chunks_with_binary_search;
-          return matches;
+void ColumnIsNullTableScanImpl::_scan_non_reference_segment(
+    const AbstractSegment& segment, const ChunkID chunk_id, RowIDPosList& matches,
+    const std::shared_ptr<const AbstractPosList>& position_filter) {
+  resolve_data_type(segment.data_type(), [&](auto type) {
+    using DataType = typename decltype(type)::type;
+
+    if (const auto* const value_segment = dynamic_cast<const BaseValueSegment*>(&segment)) {
```

Comment:
We have the if/else check here (even though the call is pretty much the same for all segment types) and in the function below we check again (then via constexpr on the template type).

We not pass abstract segments here and below use the dynamic_cast in `_scan_encoded_segment`?

Reply:
And can you add a comment why RLE is missing out here?

Reply:
I'm not sure how to do this.
`_scan_encoded_segment()` is called when we got a value, dict, lz4, or frame_of_reference segment.
In all other cases, we want to fall back to `_scan_generic_segment()` or `_scan_generic_sorted_segment()`.
Moreover, `_matches_all/none()` is only implemented for these segment types.
I also don't like the double-checking, tho.